File: fixtures/violations/csharp/CyclomaticComplexity.cs
namespace Demo;

public class RouteScore
{
    public int Score(string kind, int retries, bool urgent)
    {
        var score = 0;
        if (kind == "create") score++;
        if (kind == "update") score++;
        if (kind == "delete") score++;
        if (kind == "manual") score++;
        if (kind == "batch") score++;
        if (kind == "sync") score++;
        if (retries > 0) score++;
        if (retries > 1) score++;
        if (retries > 2) score++;
        if (urgent) score++;
        return score;
    }
}

File: fixtures/green/go/cyclomatic-complexity.go
//go:build fixture

package demo

func RouteScore(kind string, retries int, urgent bool) int {
	score := map[string]int{
		"create": 1,
		"update": 1,
		"delete": 1,
		"manual": 1,
		"batch":  1,
		"sync":   1,
	}[kind]
	if urgent {
		score++
	}
	if retries > 0 {
		score += min(retries, 3)
	}
	return score
}
